jerry-main/fuzzilli.c

```diff
@@ -0,0 +1,371 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript.h"
+#include "jerryscript-ext/handler.h"
+
+/**
+ * Maximum size of source code
+ */
+#define JERRY_BUFFER_SIZE (1048576)
+static uint8_t buffer[JERRY_BUFFER_SIZE];
+
+#include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <sys/mman.h>
+
+#define REPRL_CRFD 100
+#define REPRL_CWFD 101
+#define REPRL_DRFD 102
+#define REPRL_DWFD 103
+
+#define SHM_SIZE 0x100000
+#define MAX_EDGES ((SHM_SIZE - 4) * 8)
+
+#ifndef DCHECK
+#define DCHECK(condition) \
+  {                       \
+    assert(condition);    \
+    abort();              \
+  }
+#endif
+
+#ifndef CHECK
+#define CHECK DCHECK
+#endif
+
+FILE *fdopen(int fd, const char *mode);
+void __sanitizer_cov_reset_edgeguards(void);
```

Comment:
I changed the function name with underscores and used the `static` keyword. This was the convention used by the Fuzzilli fuzzer (I wrote only this integration).

For Clang I saw the default option `strict-prototypes` in your `CMakeLists.txt` and that's why I kept these declarations, otherwise my build failed (e.g. invoked with `CC=clang python tools/build.py --fuzzilli=on --lto=off`). What do you recommend to do instead, instead of keeping these declarations?